### PR TITLE
Migrate `extra-filename-with-temp-outputs` and `issue-85019-moved-src-dir` `run-make` tests to rmake

### PR DESCRIPTION
**Pull request by Oneirical**
Part of #121876 and the associated [Google Summer of Code project](https://blog.rust-lang.org/2024/05/01/gsoc-2024-selected-projects.html).

Please try:

try-job: armhf-gnu
// try-job: test-various // already tried
try-job: x86_64-msvc
try-job: aarch64-apple

### CODE REVIEW

**Comment by jieyouxu** [comment]
I'll take a look at this PR a bit later today.
@bors delegate+ (for try jobs)

**Review by jieyouxu — APPROVED** [review]
Looks good! r=me if try jobs come back green.

**Review by jieyouxu — APPROVED** [review]
Looks good! r=me if try jobs come back green.

File: tests/run-make/extra-filename-with-temp-outputs/rmake.rs [hunk 1]
```diff
@@ -0,0 +1,23 @@
+// In order to prevent temporary files from overwriting each other in parallel
+// compilation, rustc was changed to mix an extra filename with temporary
+// outputs. However, as this is a similar behavior with the codegen flag
+// -C extra-filename, this test checks that the manually passed flag
+// is not overwritten by this feature, and that the output files
+// are named as expected.
+// See https://github.com/rust-lang/rust/pull/15686
+
+//FIXME(Oneirical): ignore-cross-compile
```
Comment (jieyouxu) [inline tests/run-make/extra-filename-with-temp-outputs/rmake.rs:9]: Suggestion: remove this FIXME if try job comes back green since it seems to work

**Comment by Oneirical** [comment]
I have a strong suspicion that `target(target())` is breaking things again. @bors try

**Comment by Oneirical** [comment]
There we go. @rustbot review

File: src/tools/run-make-support/src/lib.rs [hunk 2]
```diff
@@ -303,6 +303,11 @@ pub fn filename_not_in_denylist<P: AsRef<Path>, V: AsRef<[String]>>(path: P, exp
         .is_some_and(|name| !expected.contains(&name.to_str().unwrap().to_owned()))
 }
 
+/// Returns true if the filename at `path` ends with `suffix`.
+pub fn has_suffix<P: AsRef<Path>>(path: P, suffix: &str) -> bool {
+    path.as_ref().file_name().is_some_and(|name| name.to_str().unwrap().ends_with(suffix))
+}
```
Comment (jieyouxu) [inline src/tools/run-make-support/src/lib.rs:309]: Question: does this addition conflict with any other PR?

File: tests/run-make/extra-filename-with-temp-outputs/rmake.rs (hunk 1 above)
Comment (jieyouxu) [reply tests/run-make/extra-filename-with-temp-outputs/rmake.rs:9]: I think this is not yet addressed.

**Review by jieyouxu — APPROVED** [review]
Thank you, just an outdated FIXME, r=me after removing it alongside a rebase.

File: src/tools/run-make-support/src/lib.rs (hunk 2 above)
Comment (Oneirical) [reply src/tools/run-make-support/src/lib.rs:309]: No other PR has `has_suffix`, it's just for this PR.

**Pull request merged**